Engine/Extensions/ExceptionExtensions.cs

```diff
@@ -1,15 +1,20 @@
+using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Text.RegularExpressions;
 
-namespace System
+namespace DotNetKoans.Engine.Extensions
 {
     public static class ExceptionExtensions
     {
-        private static readonly Regex exceptionPathRegex = new Regex(@"(\w*\.cs:line \d*)");
-
+        private static readonly Regex KoanExceptionPathRegex = new(@"(About\w*\.cs:line \d*)");
+        private static readonly Regex AssertExceptionPathRegex = new(@"\w*Asserts\.cs:line \d*(?<!About\w*\.cs:line \d*)");
```

Comment:
Huh, I never used a negative look-behind myself. Learned something today :)
We certainly can leave it like that if you want to. I just think that it is kind of unnecessary to parse the string twice and to sort of implicitly exclude the Koans here.

We could parse the string once via Regex and then separate Koan paths from the other ones.
Something like this:

```csharp
public record StackTracePaths(IEnumerable<string> KoanPaths, IEnumerable<string> OtherPaths);

private static readonly Regex ExceptionPathRegex = new(@"\w+\.cs:line \d*");
        
public static StackTracePaths GetStackTracePaths(this Exception exception)
{
	var allPaths = ExceptionPathRegex
		.Matches(exception.StackTrace ?? "").Select(x => x.Value)
		.ToList();
	
	var koanPaths = allPaths.Where(x => x.StartsWith("About")).ToList();

	return new StackTracePaths( koanPaths, allPaths.Except(koanPaths) );
}
```

Reply:
I agree that this approach would work well! I should have thought to use Records, but my college coursework is always behind on newer .NET features, so it didn't come to mind 😅

I will have an updated pull request made later tonight with these changes. I also tweaked the linq expression to exclude the list allocations.

Reply:
While it is not necessary to use a record, it is nice to have the positional syntax and the `Deconstruct` method here. We could use a tuple, but I do not like to pass tuples outside of classes. They also tend to clutter your method definitions.

> but my college coursework is always behind on newer .NET features

I do not blame them. In the last years the language is gaining features fast 😅